packages/integration-karma/test/component/LightningElement.disconnectedCallback/index.spec.js

```diff
@@ -157,3 +159,94 @@ it('should associate the component stack when the invocation throws', () => {
     expect(error.message).toBe('throw in disconnected');
     expect(error.wcStack).toBe('<x-disconnected-callback-throw>');
 });
+
+describe('disconnectedCallback for components with a custom render()', () => {
+    it('should invoke disconnectedCallback for children when parent is removed', () => {
+        let disconnectedCallbackInvoked = false;
+        function disconnectedCallback() {
+            disconnectedCallbackInvoked = true;
+        }
+        function tmpl($api, $cmp) {
+            const { c: api_custom_element, t: api_text } = $api;
+            return [
+                api_text('Parent'),
+                api_custom_element(
+                    'x-test',
+                    Test,
+                    {
+                        props: {
+                            disconnect: $cmp.callback,
+                        },
+                        key: 2,
+                    },
+                    []
+                ),
+            ];
+        }
+        registerTemplate(tmpl);
+        class Parent extends LightningElement {
+            constructor() {
+                super();
+                this.callback = disconnectedCallback;
+            }
+            render() {
+                return tmpl;
+            }
+        }
+
+        const parent = createElement('x-parent', { is: Parent });
+        document.body.appendChild(parent);
+
+        document.body.removeChild(parent);
+        expect(disconnectedCallbackInvoked).toBe(true);
+    });
+
+    xit('should invoke disconnectedCallback for children when parent switches template', () => {
+        let parentInstance;
+        let disconnectedCallbackInvoked = false;
+        function disconnectedCallback() {
+            disconnectedCallbackInvoked = true;
+        }
+        function tmpl($api, $cmp) {
+            const { c: api_custom_element, t: api_text, d: api_dynamic } = $api;
+            return [
+                api_text('Parent'),
+                api_custom_element(
+                    'x-test',
+                    Test,
+                    {
+                        props: {
+                            disconnect: $cmp.callback,
+                        },
+                        key: 2,
+                    },
+                    []
+                ),
+                api_dynamic($cmp.hideChild),
+            ];
+        }
+        registerTemplate(tmpl);
+        class Parent extends LightningElement {
+            constructor() {
+                super();
+                this.callback = disconnectedCallback;
+                this.hideChild = false;
+                parentInstance = this;
+            }
+            render() {
+                return !this.hideChild ? tmpl : () => [];
```

Comment:
@caridy I wasn't successful recreating the scenario described in https://github.com/salesforce/lwc/issues/1435#issuecomment-517407202
Updating the tracked field causes the following error message `Error: Assert Violation: markComponentAsDirty() for [object:vm undefined (3)] cannot be called during rendering of [object:vm Parent (2)].`

Reply:
we do not accept an empty template anymore... it must be a compiled template.

Reply:
the switch of the tracked template can happen in the setter of the public api.